src/lib/protocols/tls.c

```diff
@@ -470,8 +474,13 @@ static void processCertificateElements(struct ndpi_detection_module_struct *ndpi
 		       flow->protos.tls_quic_stun.tls_quic.notAfter, utcDate);
 #endif
 	      }
+	      notAfterYear = utc.tm_year;
+		  notAfterMon = utc.tm_mon;
 	    }
-
```

Comment:
This check is very complex. In the current code you have https://github.com/ntop/nDPI/blob/dev/src/lib/protocols/tls.c#L476
so it is enough to do something like
if((flow->protos.tls_quic_stun.tls_quic.notAfter-flow->protos.tls_quic_stun.tls_quic.notBefore) > threshold)
 ndpi_set_risk(flow, NDPI_TLS_CERT_VALIDITY_TOO_LONG);

Please rework this code